<launch>

    <!--args for gcs bridge for communication with QGC on Windows OS-->
    <!--https://github.com/mavlink/mavros/issues/574-->
	<!--args for airsim_node.launch, set arg host to WSL2 ipaddress config-->
	<arg name="output" default="screen"/>
	<arg name="publish_clock" default="true"/>
	<arg name="is_vulkan" default="true"/>
	<arg name="host" default="172.17.16.1" />
    
    <arg name="mavlink_bridge_url" default="udp://:14540@$(arg host):14557" />

	<!--Launch Airsim Node-->
	<include file="$(find airsim_ros_pkgs)/launch/airsim_node.launch">
		<arg name="output" value="screen"/>
		<arg name="publish_clock" value="true"/>
		<arg name="is_vulkan" value="true"/>
		<arg name="host" value="$(arg host)" />	
	</include>

    <!--https://answers.ros.org/question/353634/how-do-we-remap-all-of-the-input-and-ouput-topics-from-a-node-in-a-systematic-way/-->
    <node pkg="mavros" type="gcs_bridge" name="mavlink_bridge">
        <param name="gcs_url" value="$(arg mavlink_bridge_url)" />
    </node>

    <group ns="uav0">
        <arg name="fcu_url" default="udp://:14540@$(arg host):14557" />
        <arg name="gcs_url" default="udp://:14556@$(arg host):14551" />
        <arg name="tgt_system" default="1" />
        <arg name="tgt_component" default="1" />

        <include file="$(find utm)/launch/mavros_gcs_launch.launch">
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)"/>
        </include>
    </group>

    <group ns="uav1">
        <arg name="fcu_url" default="udp://:14541@$(arg host):14558" />
        <arg name="gcs_url" default="udp://:14556@$(arg host):14551"  />
        <arg name="tgt_system" default="2" />
        <arg name="tgt_component" default="1" />

        <include file="$(find utm)/launch/mavros_gcs_launch.launch">
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)"/>
        </include>

    </group>

    <group ns="uav2">
        <arg name="fcu_url" default="udp://:14542@$(arg host):14559" />
        <arg name="gcs_url" default="udp://:14556@$(arg host):14551"  />
        <arg name="tgt_system" default="3" />
        <arg name="tgt_component" default="1" />

        <include file="$(find utm)/launch/mavros_gcs_launch.launch">
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)"/>
        </include>
    </group>

    <group ns="uav3">
        <arg name="fcu_url" default="udp://:14543@$(arg host):14560" />
        <arg name="gcs_url" default="udp://:14556@$(arg host):14551"  />
        <arg name="tgt_system" default="4" />
        <arg name="tgt_component" default="1" />

        <include file="$(find utm)/launch/mavros_gcs_launch.launch">
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)"/>
        </include>
    </group>

    <group ns="uav4">
        <arg name="fcu_url" default="udp://:14544@$(arg host):14561" />
        <arg name="gcs_url" default="udp://:14556@$(arg host):14551"  />
        <arg name="tgt_system" default="5" />
        <arg name="tgt_component" default="1" />

        <include file="$(find utm)/launch/mavros_gcs_launch.launch">
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)"/>
        </include>
    </group>

    <group ns="uav5">
        <arg name="fcu_url" default="udp://:14545@$(arg host):14562" />
        <arg name="gcs_url" default="udp://:14556@$(arg host):14551"  />
        <arg name="tgt_system" default="6" />
        <arg name="tgt_component" default="1" />

        <include file="$(find utm)/launch/mavros_gcs_launch.launch">
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)"/>
        </include>
    </group>

    <group ns="uav6">
        <arg name="fcu_url" default="udp://:14546@$(arg host):14563" />
        <arg name="gcs_url" default="udp://:14556@$(arg host):14551"  />
        <arg name="tgt_system" default="7" />
        <arg name="tgt_component" default="1" />

        <include file="$(find utm)/launch/mavros_gcs_launch.launch">
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)"/>
        </include>
    </group>

</launch>